src/libraries/System.Private.DataContractSerialization/src/System/Xml/EncodingStreamWrapper.cs

```diff
@@ -593,14 +593,17 @@ public override int ReadByte()
             return _byteBuffer[0];
         }
 
-        public override int Read(byte[] buffer, int offset, int count)
+        public override int Read(byte[] buffer, int offset, int count) =>
+            Read(new Span<byte>(buffer, offset, count));
+
+        public override int Read(Span<byte> buffer)
         {
             try
             {
                 if (_byteCount == 0)
                 {
                     if (_encodingCode == SupportedEncoding.UTF8)
-                        return _stream.Read(buffer, offset, count);
+                        return _stream.Read(buffer);
```

Comment:
However, EncodingStreamWrapper isn't guaranteed to be a `BufferedStream` - I'm not sure why these two encoding wrappers differ. The `_stream` here is the Stream that was passed into the public API. So if the caller passes a `Stream` that doesn't override `Read(Span<byte>)`, this could run into the same problem.